core/src/main/java/com/adobe/aem/commons/assetshare/components/actions/download/impl/DownloadImpl.java

```diff
@@ -55,18 +58,27 @@
     @Default(values = "Assets")
     protected String zipFileName;
 
+    @ValueMapValue
+    @Optional
+    protected Long maxContentSize;
```

Comment:
yup thought about that. for 6.3/6.4 customers this configuration doesn't exist yet. (the updated AssetDownloadServlet is being backported to 6.3/6.4 but would require the latest service pack and doesn't exist yet...)

Reply:
It looks like it’s set as an osgi property - can we read that property? If the property doesn’t exist then, no limit (?).
We should be able to check along the way if the service ref is null, and if the property doesn’t exist (for 6.3/6.4).. wdyt?

Reply:
It looks like it’s set as an osgi property - can we read that property? If the property doesn’t exist then, no limit (?).
We should be able to check along the way if the service ref is null, and if the property doesn’t exist (for 6.3/6.4).. wdyt?

Reply:
yeah my thinking at the time was for 6.3/6.4 customers could still set a UI limit (w/ out the latest AssetDownloadServlet) but TBH they should just update as soon as its available since this is really just a UI limit on the download size. Thanks @davidjgonzalez !